### PR TITLE
[network] Consolidate repeated code in ConnMgr tests

### PR DESCRIPTION
**Pull request by gregnazario**
There was a ton of differing and repeated code for setup and other things
in the conn mgr.  This attempts to rid the code of some of the old
paradigms prior to the `Peer` object.

Just some work I did trying to understand why the tests were failing for a different PR (and then found out that my changes were definitely a problem).  Additionally, these tests actually failed more on my changes, which could be some side effects of me removing the `eligible` peer part.  However, all of these tests run successfully, and should have less randomness in them.

### CODE REVIEW

File: network/src/connectivity_manager/test.rs [hunk 1]
```diff
@@ -23,25 +23,54 @@ const MAX_TEST_CONNECTIONS: usize = 3;
...
-    let addr = NetworkAddress::from_str("/ip4/127.0.0.1/tcp/9090").unwrap();
-    let addr = addr.append_prod_protos(pubkey, diem_config::config::HANDSHAKE_VERSION);
-    (peer_id, pubkey, pubkeys, addr)
+    let addr = network_address_with_pubkey(addr_str, pubkey);
+    (
+        peer_id,
+        Peer::new(vec![addr.clone()], pubkeys, PeerRole::Validator),
+        pubkey,
+        addr,
+    )
```
Comment (davidiw) [inline network/src/connectivity_manager/test.rs:65]: so in one case you perform an op in place (Peer:new) and in other cases you define intermediate values, save tre.. err lines ;)

**Review by davidiw — APPROVED** [review]
seems useful :D

**Review by phlip9 — APPROVED** [review]
awesome! this is definitely something I noticed when I refactored these; glad we could finally clean this up : )

**Pull request merged**